test/js/source/tile_pyramid.test.js

```diff
@@ -648,3 +648,31 @@ test('TilePyramid#orderedIDs (ascending order by zoom level)', function(t) {
     ]);
     t.end();
 });
+
+
+test('TilePyramid#findLoadedParent', function(t) {
+    var pyramid = createPyramid({});
+    var tr = new Transform();
+    tr.width = 512;
+    tr.height = 512;
+    pyramid.updateCacheSize(tr);
+
+    var tile = {
+        coord: new TileCoord(1, 0, 0),
+        loaded: true
+    };
+    pyramid._cache.add(tile.coord.id, tile);
+
+    t.equal(pyramid.findLoadedParent(new TileCoord(2, 3, 3), 0, {}), undefined);
+
+    // add from cache
+    t.deepEqual(pyramid.findLoadedParent(new TileCoord(2, 0, 0), 0, {}), tile);
+
+    // cache is empty
+    t.equal(pyramid._cache.order.length, 0);
+
+    // add from retained tiles
+    t.deepEqual(pyramid.findLoadedParent(new TileCoord(2, 0, 0), 0, {}), tile);
```

Comment:
`findLoadedParent` looks for load parent tiles in both the cache and `._tiles` (the previous set of tiles).
669 tests whether it works when the tile is in the cache
675 tests whether it works when the tile is in `._tiles`
669 moves the tile from the cache to ._tiles internally which is what 672 confirms.

I'll split it up as suggested by https://github.com/mapbox/mapbox-gl-js/pull/2168/files#r53679813